### PR TITLE
Unnest medical facility type and name

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
Unnest medical facility and name so the data is not dependent on each other. This way the API will be able to handle each piece of data separately and not have both questions optionally required.  

## Testing done
- End to end testing
- Manual testing
- Unit testing

## Screenshots
image

## Acceptance criteria
- [x] Remove the optional require from facility name & type
- [x] Make it so API accepts both facility name & type separately 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by LinnJS** [comment]
This work is dependent on department-of-veterans-affairs/va.gov-team#16477

**Review by Mottie — APPROVED** [review]

File: src/applications/caregivers/helpers.js [hunk 1]
```diff
@@ -161,14 +161,14 @@ export const validateSSNIsUnique = (errors, formData) => {
...
-export const facilityNameMaxLength = (errors, formData) => {
-  const facilityNameLength = formData.veteranLastTreatmentFacility.name?.length;
-  if (facilityNameLength > 80) {
-    errors.addError(
-      "You've entered too many characters, please enter less than 80 characters.",
-    );
-  }
-};
+// export const facilityNameMaxLength = (errors, formData) => {
+//   const facilityNameLength = formData.veteranLastTreatmentFacilityName?.length;
```
Comment (LinnJS) [inline src/applications/caregivers/helpers.js:165]: need to remove this

**Review by craigwheeler — APPROVED** [review]

**Comment by LinnJS** [comment]
Blocked by backend breaking changes

**Comment by LinnJS** [comment]
Closing this PR as there has been multiple refactors since this PR was opened and would be easier to reimplement than resolve conflicts.

**Pull request closed without merging**